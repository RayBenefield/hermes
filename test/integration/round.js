import flame from '@leonardvandriel/flame';
import * as databases from './database'; // eslint-disable-line import/no-unresolved,import/extensions,no-unused-vars
import uuid from '../utils/mock-uuid';
import random from '../utils/mock-random';
import configureSave from '../utils/save-db';
import configureDomain from '../../src/domain';
import configureFlame from '../../src/lib/local-db';

const db = configureFlame(flame);
const domain = configureDomain({ db, uuid, random });
const save = configureSave(flame); // eslint-disable-line no-unused-vars
beforeEach(() => flame.loadDatabase(undefined));

test('notify all players of the new round', () => {
    // Given
    flame.loadDatabase(databases['new-round']);

    // When
    return domain({
        action: 'round',
        round: {
            id: 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa',
            game: 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa',
        },
    })
        .then(results => expect({
            results,
            db: flame.get('/'),
        }).toMatchSnapshot());
});
