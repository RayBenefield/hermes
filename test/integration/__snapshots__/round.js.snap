// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notify all players of the new round 1`] = `
Object {
  "db": Object {
    "games": Object {
      "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
        "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        "notified_players": Object {
          "123456": true,
          "234567": true,
          "345678": true,
          "456789": true,
          "567890": true,
        },
        "players": Object {
          "123456": true,
          "234567": true,
          "345678": true,
          "456789": true,
          "567890": true,
        },
        "rounds": Object {
          "--------------------": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        },
      },
    },
    "hands": Object {
      "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
        "123456": Object {
          "cards": Object {
            "0": true,
            "519": true,
            "520": true,
            "521": true,
            "522": true,
            "524": true,
            "525": true,
            "526": true,
            "527": true,
            "528": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 123456,
        },
        "234567": Object {
          "cards": Object {
            "0": true,
            "259": true,
            "260": true,
            "261": true,
            "262": true,
            "263": true,
            "783": true,
            "784": true,
            "786": true,
            "787": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 234567,
        },
        "345678": Object {
          "cards": Object {
            "0": true,
            "130": true,
            "131": true,
            "392": true,
            "393": true,
            "653": true,
            "654": true,
            "655": true,
            "915": true,
            "917": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 345678,
        },
        "456789": Object {
          "cards": Object {
            "0": true,
            "196": true,
            "327": true,
            "458": true,
            "589": true,
            "720": true,
            "850": true,
            "851": true,
            "981": true,
            "982": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 456789,
        },
        "567890": Object {
          "cards": Object {
            "0": true,
            "1014": true,
            "163": true,
            "19": true,
            "294": true,
            "32": true,
            "687": true,
            "818": true,
            "949": true,
            "98": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 567890,
        },
      },
    },
    "players": Object {
      "facebook": Object {
        "123456": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 123456,
          "platform": "facebook",
        },
        "234567": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 234567,
          "platform": "facebook",
        },
        "345678": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 345678,
          "platform": "facebook",
        },
        "456789": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 456789,
          "platform": "facebook",
        },
        "567890": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 567890,
          "platform": "facebook",
        },
      },
    },
    "queue": Object {},
    "rounds": Object {
      "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
        "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
          "card": Object {
            "contents": "_?  There's an app for that.",
            "id": 0,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        },
      },
    },
  },
  "results": Object {
    "action": "round",
    "card": Object {
      "contents": "_?  There's an app for that.",
      "id": 0,
    },
    "game": Object {
      "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
      "notified_players": Object {
        "123456": true,
        "234567": true,
        "345678": true,
        "456789": true,
        "567890": true,
      },
      "players": Object {
        "123456": true,
        "234567": true,
        "345678": true,
        "456789": true,
        "567890": true,
      },
    },
    "messages": Array [
      Object {
        "payload": Object {
          "card": Object {
            "contents": "_?  There's an app for that.",
            "id": 0,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "round": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        },
        "type": "new-goal-message",
      },
    ],
    "players": Array [
      Object {
        "first_name": "Ray",
        "games": Object {
          "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
        },
        "id": 123456,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "games": Object {
          "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
        },
        "id": 234567,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "games": Object {
          "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
        },
        "id": 345678,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "games": Object {
          "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
        },
        "id": 456789,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "games": Object {
          "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
        },
        "id": 567890,
        "platform": "facebook",
      },
    ],
    "round": Object {
      "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
      "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
    },
  },
}
`;
